st2common/st2common/models/api/action.py

```diff
@@ -148,6 +149,11 @@ class ActionAPI(BaseAPI):
                 "patternProperties": {
                     "^\w+$": util_schema.get_draft_schema()
                 }
+            },
+            "tags": {
+                "description": "User associated metadata assigned to this object.",
```

Comment:
> On a related note - we should limit the maximum field size to prevent resource exhaustion, ddos and similar attacks.

Good point. How about 1K, larger than a tweet :P, characters? With utf-8 i guess that would be a max of 4KB per name and value. This makes some room to store structured data or useful text in the value field.